.changeset/rotten-students-agree.md

```diff
@@ -0,0 +1,5 @@
+---
+"@pnpm/config": patch
+---
+
+move use of PNPM_HOME into getDataDir
```

Comment:
you can remove the changeset. There's no reason to release this change as it doesn't change the logic.